include/GIL/Type.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef GLU_GIL_TYPE_HPP
+#define GLU_GIL_TYPE_HPP
+
+#include "TypeBase.hpp"
+
+namespace glu::gil {
+
+/// @brief Represents a type in the GIL (Glu Intermediate Language).
+/// For more information, see the documentation here:
+/// https://glu-lang.org/gil/
+class Type {
+    struct Fields {
+        unsigned char size : 48;
+        unsigned char alignment : 5;
+        unsigned char isConst : 1;
```

Comment:
uint64_t as you are using 64 bits, otherwise you can use more than 8 bits with a char
Also add a comment saying 10 bits are available